src/lib/Runner.ts

```diff
@@ -459,18 +456,51 @@ export class Runner {
         return this.next();
       case 'FAILED':
         logMessage('Moving landRequest to failed state');
-        await landRequest.setStatus('fail', 'Landkid build failed');
+        await this.failAllDependents(landRequestStatus, 'fail', 'Landkid build failed');
         return this.next();
       case 'STOPPED':
-        logMessage('Moving landRequest to aborted state');
-        await landRequest.setStatus('aborted', 'Landkid pipelines build was stopped');
+        logMessage(`Moving landRequest to aborted state`);
+        await this.failAllDependents(
+          landRequestStatus,
+          'aborted',
+          'Landkid pipelines build was stopped',
+        );
         return this.next();
       default:
         logMessage('Dont know what to do with build status, ignoring');
         break;
     }
   };
 
+  // Mark the current request as failed / aborted and fail all dependents within 'status-transition' lock.
+  // So, multiple free slots are available in one go for the speculation engine to process.
+  failAllDependents = async (
+    currentLandRequestStatus: LandRequestStatus,
+    currentRequestState: IStatusState,
+    currentRequestReason: string,
```

Comment:
nit: The `current` prefix makes sense for the first parameter but I'm not sure about the other two since they represent the new state rather than the current state.

Reply:
can rename them to `state` and `reason`